assets/src/components/activities/oli_embedded/OliEmbeddedAuthoring.tsx

```diff
@@ -0,0 +1,250 @@
+import { configureStore } from 'state/store';
+import React from 'react';
+import {
+  AuthoringElement,
+  AuthoringElementProps,
+  AuthoringElementProvider,
+  useAuthoringElementContext,
+} from 'components/activities/AuthoringElement';
+import { OliEmbeddedModelSchema } from 'components/activities/oli_embedded/schema';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import * as ActivityTypes from 'components/activities/types';
+import { OliEmbeddedActions } from 'components/activities/oli_embedded/actions';
+
+import guid from 'utils/guid';
+import { uploadFiles } from 'components/media/manager/upload';
+import { CloseButton } from 'components/misc/CloseButton';
+import { MediaItemRequest, ScoringStrategy } from 'components/activities/types';
+import { lastPart } from 'components/activities/oli_embedded/utils';
+import { ActivityXmlEditor } from 'components/common/ActivityXmlEditor';
+import ModalSelection from 'components/modal/ModalSelection';
+const store = configureStore();
+
+const Embedded = (props: AuthoringElementProps<OliEmbeddedModelSchema>) => {
+  const { dispatch, model, onRequestMedia } = useAuthoringElementContext<OliEmbeddedModelSchema>();
+
+  const { projectSlug } = props;
+
+  function select(_projectSlug: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const request = {
+        type: 'MediaItemRequest',
+      } as MediaItemRequest;
+      if (props.onRequestMedia) {
+        props.onRequestMedia(request).then((r) => {
+          if (r === false) {
+            reject('error');
+          } else {
+            resolve(r as string);
+          }
+        });
+      }
+    });
+  }
+
+  const addFile = (e: any) => {
+    select(projectSlug).then((url: string) => {
+      dispatch(OliEmbeddedActions.addResourceURL(url));
+    });
+  };
+
+  const onFileUpload = (files: FileList) => {
+    // get a list of the files to upload
+    const fileList: File[] = [];
+    for (let i = 0; i < files.length; i = i + 1) {
+      const file = files[i];
+      fileList.push(renameFile(file, 'webcontent/' + model.resourceBase + '/' + file.name));
+    }
+
+    uploadFiles(projectSlug, fileList)
+      .then((result: any) => {
+        result.forEach((i: any) => {
+          dispatch(OliEmbeddedActions.addResourceURL(i.url));
+        });
+      })
+      .catch((reason: any) => {
+        console.log(JSON.stringify(reason.message));
```

Comment:
might not need that considering you're going to display the modal

Reply:
Console message removed